src/middleware.ts

```diff
@@ -1,6 +1,11 @@
 import { HttpTypes } from "@medusajs/types"
 import { NextRequest, NextResponse } from "next/server"
 
+import createIntlMiddleware from "next-intl/middleware"
+import { fallbackLng, intlConfig, languages } from "./lib/i18n/settings"
+
+const intlMiddleware = createIntlMiddleware(intlConfig)
```

Comment:
Currently, routing is done based on a pattern like `[locale]/[countryCode]`.

For e-commerce stores, it's common to have either only the country code or optionally a locale following the country (e.g. `/uk`, `/ch/fr`, …).

I'd recommend to use [custom prefixes](https://next-intl.dev/docs/routing#locale-prefix-custom) in order to map a country code to a locale that can be used internally.

You'd have to change the currently used `[countryCode]` segment to `[locale]`, but the public routes would stay the same. Additionally, you can retrieve the country code from the locale in your app code:

```tsx
const locale = new Intl.Locale('fr-CH');
locale.region; // 'CH')
```